### PR TITLE
[GCE] Support Managed Instance Groups

### PR DESCRIPTION
**Pull request by supertom**
### Description

This PR aims to add support for Managed Instance Groups (MIGs) on GCE.  Basic support for AutoScalers has also been added.

The following methods have been added (with supporting classes).
- gce.ex_get_instancetemplate(template)
- gce.ex_list_instancetemplates()
- gce.ex_get_instancegroupmanager(name, zone)
- gce.ex_list_instancegroupmanagers()
- gce.ex_create_instancegroupmanager(name, zone, template, size)
- gce.ex_destroy_instancegroupmanager(instance_group)
- gce.ex_instancegroupmanager_resize(instance_group, num_instances)
- gce.ex_instancegroupmanager_recreate_instances(instance_group, list_of_instances)
- gce.ex_instancegroupmanager_list_managed_instances(instance_group)
- gce.ex_get_autoscaler(as_name, as_zone)
- gce.ex_create_autoscaler(as_name, as_zone, my_instance_group, autoscaler_policy, description)
- gce.ex_list_autoscalers(as_zone)

(Some) Tests and fixtures have also been added.
#### Useful Links
- [Creating Groups of Instances on GCE](https://cloud.google.com/compute/docs/instance-groups)
- [API docs](https://cloud.google.com/compute/docs/reference/latest/instanceGroupManagers)
- [API docs (AutoScalers)](https://cloud.google.com/compute/docs/reference/latest/autoscalers)
### Example Usage

```
print gce.ex_list_autoscalers()
print gce.ex_list_instancetemplates()
print gce.ex_list_instancegroupmanagers()
print gce.ex_instancegroupmanagers_list_managed_instances()

# Create an Instance Group
ig_name = 'my-instance-group-foo7'
ig_zone = 'us-central1-c'
it_name = 'my-instance-template-1'

template = gce.ex_get_instancetemplate(it_name)
instance_group = gce.ex_create_instancegroupmanager(ig_name, ig_zone, template, 5, 'base-foo', "my description")
print instance_group

print gce.ex_instancegroupmanager_resize(instance_group, 4)
print gce.ex_instancegroupmanager_recreate_instances(instance_group)

# add autoscaling to the instance group
autoscaler_policy = {
  "maxNumReplicas": 7,
  "minNumReplicas": 4,
  "coolDownPeriodSec": 35,
  "cpuUtilization": {"utilizationTarget": .66}
}

new_as = gce.ex_create_autoscaler(
  as_name, as_zone, my_instance_group, autoscaler_policy,
  description='My desc for this autoscaler')
print new_as
print gce.ex_list_autoscalers(as_zone)

# Update AutoScaler
updated_as = gce.ex_get_autoscaler(as_name, as_zone)
updated_as.policy['coolDownPeriodSec'] = 71
updated_as.policy['cpuUtilization']['utilizationTarget'] = .14
print gce.ex_update_autoscaler(updated_as)

# Delete the Instance Group.  VMs are also deleted.
deleted = gce.ex_destroy_instancegroupmanager(instance_group)
print deleted # returns True
```
### Status
- ready_for_review
### Checklist (tick everything that applies)
- [X] [Code linting](http://libcloud.readthedocs.org/en/latest/development.html#code-style-guide) (required, can be done after the PR checks)
- [X] Documentation
- [X] [Tests](http://libcloud.readthedocs.org/en/latest/testing.html)
- [X] [ICLA](http://libcloud.readthedocs.org/en/latest/development.html#contributing-bigger-changes) (required for bigger changes)

### CODE REVIEW

**Comment by tonybaloney** [comment]
Good work Tom. Do you have an ICLA signed btw?

File: libcloud/compute/drivers/gce.py [hunk 1]
```diff
@@ -751,6 +751,95 @@ def __repr__(self):
...
+    def __init__(self, id, name, zone, size, template, driver, extra=None):
+        """
+        :param  id: Internal identifier of Instance Group.  Display only.
+        :type   id: ``str``
+
+        :param  name: The name of this Instance Group.
+        :type   size: ``str``
+
+        :param  zone: Zone to wich the Instance Group belongs
+        :type   zone: ``str``
```
Comment (erjohnso) [inline libcloud/compute/drivers/gce.py]: Consider also allowing `GCEZone`. Same with other methods.

**Comment by erjohnso** [comment]
Small nit about allowing other param types, but otherwise LGTM!

**Comment by supertom** [comment]
@tonybaloney I thought I did have an ICLA on file, but I didn't, so thanks for asking.  It's now on file with Apache.

**Comment by tonybaloney** [comment]
That's cool @supertom, I don't have access to check them. I won't ask again!

**Pull request closed without merging**